src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, null);
+        if (isEmpty()) {
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        Node<T> newNode;
+        if (index == 0) {
+            newNode = new Node<>(value, head);
+            newNode.next = head;
+            head = newNode;
+            if (size == 0) {
+                tail = newNode;
+            }
+        } else {
+            Node<T> prevNode = findNodeByIndex(index - 1);
+            newNode = new Node<>(value, prevNode.next);
+            prevNode.next = newNode;
+            if (newNode.next == null) {
+                tail = newNode;
+            }
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T node : list) {
+            add(node);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> targetNode = findNodeByIndex(index);
```

Comment:
call checkindex inside of findnodebyindex instead of calling them together repeatedly